tests/fixed/UselessConditions.php

```diff
@@ -44,12 +44,20 @@ public function unecessaryIfMethodForEarlyReturn() : bool
 
     public function uselessIfConditionWithParameter(bool $bool) : bool
     {
-        return ! $bool;
+        if ($bool) {
+            return false;
+        }
+
+        return true;
     }
 
     public function uselessIfConditionWithBoolMethod() : bool
     {
-        return ! $this->isTrue();
+        if ($this->isTrue()) {
```

Comment:
See https://github.com/slevomat/coding-standard/issues/604 - CS doesn't know the type of `$this->isTrue()` and in general it may produce broken code (the method would return non-bool). This is now still reported, but not auto-fixed.